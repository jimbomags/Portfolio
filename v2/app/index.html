<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <title>James McGill Front End Developer</title>
  </head>
  <body>
    <nav id='nav'>

    </nav>
    <header>

    </header>
    <section id="about">

    </section>
    <section id="skills">

    </section>
    <section id="portfolio">

    </section>
    <section id="contact">

    </section>
    <footer>

    </footer>
  </body>
  <script type="text/javascript" src="./src/bundle.js"></script>
</html>
