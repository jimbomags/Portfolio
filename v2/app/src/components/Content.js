import React from 'react'
import Contact from './Contact'
import Weather from './Weather'

const Content = {
  home:
    <h1>James McGill<br/>Frontend Developer</h1>,
  about:
    <span id='about'>
      <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
      <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
    </span>,
  skills: [
    [1,'./assets/html5.png',<p>HTML</p>],
    [2,'./assets/css3.png',<p>CSS</p>],
    [3,'./assets/javascript.png',<p>Javascript</p>],
    [4,'./assets/react.jpg',<p>React</p>],
    [5,'./assets/node.png',<p>Node</p>],
    [6,'./assets/sass.png',<p>Sass</p>]
  ],
  contact: <Contact />,
  portfolio: {
    weather: <Weather />
  }
}

export default Content
