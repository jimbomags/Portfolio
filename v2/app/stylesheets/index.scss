@import './modules/nav-bar.scss';
@import './modules/portfolio.scss';
@import './modules/mobile.scss';
@import './modules/home.scss'
