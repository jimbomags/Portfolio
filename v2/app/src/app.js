import React, { Component } from 'react'
import Content from './components/Content'

const skills = <div id='skills_container'>
  {Content.skills.map((item) => {
    return (
      <div key={item[0]} className='skills'>
        <img src={item[1]} />
        {item[2]}
      </div>
    )
  })}
</div>

class App extends Component {
  constructor(props) {
    super(props)
    this.state = {
      content: Content.home
    }
    this.changeContent = this.changeContent.bind(this)
    this.portClicked = this.portClicked.bind(this)
    this.displayNav = this.displayNav.bind(this)
  }
  changeContent(newContent) {
    const contContainer = document.querySelector('#content')
    if (newContent == skills) {
      contContainer.style.backgroundImage = 'none'
    }
    else if (this.state.content == skills) {
      contContainer.style.backgroundImage = 'url(\'/assets/background_desktop.jpg\')'
    }
    this.setState({content: newContent})
  }
  portClicked() {
    const port = document.querySelectorAll('.port')

    if (window.screen.width < 1050) {

      port.forEach(item => {
        if (item.style.display == 'block') {
          item.style.display = 'none'
        }
        else {
          item.style.display = 'block'
        }
        item.classList.toggle('clicked_port')
      })
    }
  }
  displayNav() {
    const links = document.querySelector('#links')
    if (links.style.display == 'flex') {
      links.style.display = ''
    }
    else {
      links.style.display = 'flex'
    }
  }

  render() {
    return (
      <div id='main_container'>
        <nav>
          <img id='logo' src='../assets/logo.png' onClick={() => {
            this.changeContent(Content.home)}
          }/>
          <ul id='links'>
            <p onClick={() => {
              this.displayNav()
              this.changeContent(Content.home)}
            }>Home</p>
            <p onClick={() => {
              this.displayNav()
              this.changeContent(Content.about)}
            }>About</p>
            <p onClick={() => {
              this.displayNav()
              this.changeContent(skills)}
            }>Skills</p>
            <div id='port_links'>
              <p onClick={this.portClicked}>Portfolio</p>
              <div id='port_links_content'>
                <p className='port' onClick={() => {
                  this.displayNav()
                  this.changeContent(Content.portfolio.rndQuote)}
                }>Random Quote</p>
                <p className='port' onClick={() => {
                  this.displayNav()
                  this.changeContent(Content.portfolio.weather)}
                }>Weather App</p>
                <p className='port' onClick={() => {
                  this.displayNav()
                  this.changeContent(Content.portfolio.sandl)}
                }>Snakes & Ladders</p>
              </div>
            </div>
            <p onClick={() => {
              this.displayNav()
              this.changeContent(Content.contact)}
            }>Contact</p>
          </ul>
          <div id='burger_icon' onClick={this.displayNav}>
            <div></div>
            <div></div>
            <div></div>
          </div>
        </nav>
        <div id='content'>
          {this.state.content}
        </div>
      </div>
    )
  }
}

export default App
