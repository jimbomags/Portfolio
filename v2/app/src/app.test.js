import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { App, Title, About } from './app';
import Content from './components/Content';

configure({ adapter: new Adapter() })

test('It should render', () => {
  const component = renderer.create(<App />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Portfolio links should appear/disappear when burger is clicked', () => {
  const wrapper = mount(<App />);

  wrapper.find('#burger_icon').simulate('click')

  expect(wrapper.find('#links').prop('style')).toEqual({ 'display': 'flex' })

  wrapper.find('#burger_icon').simulate('click')

  expect(wrapper.find('#links').prop('style')).toEqual({ 'display': 'none' })

});

test('Title should render the property passed to it', () => {
  const wrapper = shallow(<Title title={Content.title} />)

  expect(wrapper.find('h1').text()).toEqual('James McGillFrontend Developer')
  expect(wrapper.find('a').prop('href')).toEqual('#about')
});

test('About should render the property passed to it', () => {
  const wrapper = shallow(<About about={Content.about} />)

  expect(wrapper.html()).toMatch(/[(<p>)(junior)]/);
});
