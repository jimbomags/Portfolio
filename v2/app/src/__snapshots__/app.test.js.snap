// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render 1`] = `
<div
  id="whole_container"
>
  <div
    id="home_container"
  >
    <nav>
      <div
        id="burger_icon"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        <div />
        <div />
        <div />
      </div>
      <ul
        id="links"
        style={
          Object {
            "display": "",
          }
        }
      >
        <a
          href="#about"
          tabIndex="0"
        >
          About
        </a>
        <a
          href="#skills"
          role="button"
          tabIndex="0"
        >
          Skills
        </a>
        <a
          href="#portfolio-container"
          role="button"
          tabIndex="0"
        >
          Portfolio
        </a>
        <a
          href="#contact"
          role="button"
          tabIndex="0"
        >
          Contact
        </a>
      </ul>
    </nav>
    <div
      id="home"
    >
      <h1>
        James McGill
        <br />
        <span>
          Frontend Developer
        </span>
      </h1>
      <a
        href="#about"
      >
        <button
          id="about-me-btn"
        >
          Find Out More
        </button>
      </a>
    </div>
  </div>
  <div
    id="about"
  >
    <div>
      <h3>
        Why I Started...
      </h3>
      <p>
        Hi, I’m James, and I’m a self-taught Junior Front End Developer. I started learning to code in August 2016 after deciding to change my career. I began by looking at my interests for inspiration and two things stood out, video games and tech. I immediately ruled out playing video games as I normally get thrashed when I play online – so I looked into tech. In deciding what to learn I looked back to when I was studying Computing for my A-levels. I was led to choosing web development when I realised my favourite projects were building a simple website and coding a snakes & ladders game in BASIC.
      </p>
      <h3>
        How I Started...
      </h3>
      <p>
        Not knowing where to start I searched 'how to be a web developer' and started learning Python from a recommendation in a blog. Shortly after starting this course I became disgruntled, I was copying code to a code editor and not learning much. Therefore, I researched other options and decided to switch to learning HTML & CSS from Codecademy & Lynda - although I still plan on learning Python in the future.
      </p>
      <p>
        Once I had an understanding of HTML & CSS I moved onto JavaScript. Initially I used Codecademy and Lynda but was struggling to learn from their JS courses. That led me to search other options and I found FreeCodeCamp. I liked FCC because you learn by writing code to complete challenges. You start small such as declaring variables before moving onto more complex algorithms. Another part of FCC’s syllabus is completing projects, which comprised my first 
        <a
          href="https://decorous-glove.surge.sh"
          target="_blank"
        >
          portfolio.
        </a>
      </p>
      <h3>
        Where I Am Now...
      </h3>
      <p>
        When I finished my portfolio I decided to get feedback on my skills and what I needed to improve. The advice I received and from looking at job adverts, were that I needed to learn ES6, Node.js, a CSS pre-processor, task runners and a JS framework. At time of writing, the FCC syllabus is great for learning vanilla ES5 JavaScript but doesn’t provide very good content for the above skills. Therefore, I switched back to Lynda & Udemy. I watched a series of video courses to learn the above – choosing React for the JS framework – and deciding the best way to hone these skills was to build something with them. I chose to re-make my portfolio using React and re-creating two of the projects from my original portfolio. Finally, using my computing A-level as inspiration, I challenged to myself to create a snakes & ladders game. I hope you like it.
      </p>
    </div>
  </div>
  <div
    id="skills"
  >
    <h2>
      My Skill Set...
    </h2>
    <div
      id="skill-images-container"
    >
      <div
        className="skills"
      >
        <img
          alt={
            <p>
              HTML
            </p>
          }
          src="./assets/html5.png"
        />
        <p>
          HTML
        </p>
      </div>
      <div
        className="skills"
      >
        <img
          alt={
            <p>
              CSS
            </p>
          }
          src="./assets/css3.png"
        />
        <p>
          CSS
        </p>
      </div>
      <div
        className="skills"
      >
        <img
          alt={
            <p>
              Javascript
            </p>
          }
          src="./assets/javascript.png"
        />
        <p>
          Javascript
        </p>
      </div>
      <div
        className="skills"
      >
        <img
          alt={
            <p>
              React
            </p>
          }
          src="./assets/react.jpg"
        />
        <p>
          React
        </p>
      </div>
      <div
        className="skills"
      >
        <img
          alt={
            <p>
              Node
            </p>
          }
          src="./assets/node.png"
        />
        <p>
          Node
        </p>
      </div>
      <div
        className="skills"
      >
        <img
          alt={
            <p>
              Sass
            </p>
          }
          src="./assets/sass.png"
        />
        <p>
          Sass
        </p>
      </div>
    </div>
  </div>
  <div
    id="portfolio-container"
  >
    <div
      id="back"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      style={
        Object {
          "display": "",
        }
      }
      tabIndex="0"
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m36.3 20v2.9q0 1.1-0.7 2t-1.9 0.8h-15.7l6.5 6.6q0.9 0.8 0.9 2t-0.9 2l-1.7 1.7q-0.8 0.8-2 0.8-1.1 0-2-0.8l-14.5-14.6q-0.9-0.8-0.9-2 0-1.1 0.9-2l14.5-14.5q0.8-0.9 2-0.9 1.2 0 2 0.9l1.7 1.6q0.9 0.9 0.9 2.1t-0.9 2l-6.5 6.5h15.7q1.1 0 1.9 0.9t0.7 2z"
          />
        </g>
      </svg>
      Back
    </div>
    <div
      id="portfolio-home"
    >
      <h2>
        What I can do...
      </h2>
      <div
        id="port-links-container"
      >
        <div
          className="link-container"
        >
          <div
            className="port-links"
            id="port-sl"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
          />
          <div
            className="port-info"
          >
            <h3>
              Snakes & Ladders Game
            </h3>
          </div>
        </div>
        <div
          className="link-container"
        >
          <div
            className="port-links"
            id="port-weather"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
          />
          <div
            className="port-info"
          >
            <h3>
              Weather App
            </h3>
          </div>
        </div>
        <div
          className="link-container"
        >
          <div
            className="port-links"
            id="port-quote"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
          />
          <div
            className="port-info"
          >
            <h3>
              Random Quote Generator
            </h3>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    id="contact"
  >
    <h2>
      How to reach me...
    </h2>
    <div
      id="contact-container"
    >
      <form
        action="https://formspree.io/contact@jamesmcgill.co.uk"
        id="form"
        method="POST"
        onSubmit={[Function]}
      >
        Name:
        <input
          id="name"
          name="name"
          onChange={[Function]}
          type="text"
          value=""
        />
        <span
          id="name-error"
        />
        E-mail:
        <input
          id="email"
          name="email"
          onChange={[Function]}
          type="email"
          value=""
        />
        <span
          id="email-error"
        />
        Message:
        <textarea
          id="message"
          name="message"
          onChange={[Function]}
          value=""
        />
        <span
          id="message-error"
        />
        <div
          className="submitBtn"
        >
          <button
            className="button"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
      <div
        id="contact-icons"
      >
        <a
          href="https://github.com/jimbomags"
          target="_blank"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m20.1 2.9q4.7 0 8.6 2.3t6.3 6.2 2.3 8.6q0 5.6-3.3 10.1t-8.4 6.2q-0.6 0.1-0.9-0.2t-0.3-0.7q0 0 0-1.7t0-3q0-2.1-1.2-3.1 1.3-0.2 2.3-0.4t2.1-0.9 1.8-1.5 1.2-2.3 0.5-3.4q0-2.7-1.8-4.6 0.8-2-0.2-4.5-0.6-0.2-1.8 0.2t-2 1l-0.9 0.5q-2-0.6-4.3-0.6t-4.2 0.6q-0.4-0.2-1-0.6t-1.9-0.8-1.9-0.3q-1 2.5-0.1 4.5-1.8 1.9-1.8 4.6 0 1.9 0.5 3.4t1.1 2.3 1.8 1.5 2.1 0.9 2.3 0.4q-0.9 0.8-1.1 2.3-0.4 0.2-1 0.3t-1.3 0.1-1.4-0.5-1.3-1.4q-0.4-0.7-1-1.1t-1.1-0.6l-0.5 0q-0.5 0-0.6 0.1t-0.1 0.2 0.2 0.3 0.2 0.3l0.2 0.1q0.5 0.2 1 0.9t0.7 1.1l0.2 0.5q0.3 0.9 1 1.4t1.5 0.7 1.5 0.1 1.3-0.1l0.5 0q0 0.8 0 1.9t0 1.2q0 0.5-0.3 0.7t-0.9 0.2q-5.2-1.7-8.4-6.2t-3.3-10.1q0-4.7 2.3-8.6t6.2-6.2 8.6-2.3z m-10.6 24.6q0.1-0.2-0.2-0.3-0.2-0.1-0.2 0.1-0.1 0.1 0.1 0.2 0.2 0.2 0.3 0z m0.7 0.7q0.1-0.1-0.1-0.3-0.2-0.2-0.3-0.1-0.2 0.1 0 0.4 0.3 0.2 0.4 0z m0.7 1q0.2-0.1 0-0.4-0.2-0.3-0.4-0.1-0.2 0.1 0 0.4t0.4 0.1z m0.9 1q0.2-0.2-0.1-0.4-0.3-0.3-0.4-0.1-0.2 0.2 0 0.4 0.3 0.3 0.5 0.1z m1.3 0.5q0-0.2-0.3-0.3-0.4-0.1-0.4 0.1t0.2 0.4q0.4 0.1 0.5-0.2z m1.4 0.1q0-0.2-0.4-0.2-0.4 0-0.4 0.2 0 0.3 0.4 0.3 0.4 0 0.4-0.3z m1.3-0.2q-0.1-0.2-0.4-0.2-0.4 0.1-0.3 0.4t0.4 0.1 0.3-0.3z"
              />
            </g>
          </svg>
        </a>
        <a
          href="mailto:contact@jamesmcgill.co.uk"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m40 15.8v17.8q0 1.4-1 2.5t-2.6 1h-32.8q-1.5 0-2.6-1t-1-2.5v-17.8q1 1.1 2.3 2 8 5.5 11 7.7 1.3 0.9 2.1 1.5t2.1 1 2.5 0.6h0q1.2 0 2.5-0.6t2.1-1 2.1-1.5q3.7-2.8 11.1-7.7 1.2-0.9 2.2-1.9z m0-6.5q0 1.7-1.1 3.4t-2.7 2.7q-8.4 5.8-10.5 7.3-0.2 0.1-0.9 0.6t-1.2 0.9-1.2 0.7-1.3 0.6-1.1 0.2h0q-0.5 0-1.1-0.2t-1.3-0.6-1.2-0.7-1.2-0.9-0.9-0.6q-2.1-1.5-5.9-4.1t-4.6-3.2q-1.3-0.9-2.6-2.6t-1.2-3q0-1.8 0.9-2.9t2.7-1.2h32.8q1.5 0 2.5 1.1t1.1 2.5z"
              />
            </g>
          </svg>
        </a>
      </div>
    </div>
  </div>
</div>
`;
